### PR TITLE
Remove 'pubspec_overrides.yaml' when running 'pub upgrade' or 'pub outdated'.

### LINKED ISSUE #1089: pana does not handle `pubspec_overrides.yaml`
`freezed` uses a **pubspec_overrides.yaml** file, that overrides to the local `freezed_annotation` during development
```yaml
dependency_overrides:
  freezed_annotation:
    path: ../freezed_annotation
```

But this results in `pana --hosted freezed` to not be able to resolve those dependencies during **✗ Pass static analysis**.

### How to reproduce:
```sh
dart pub global activate pana
pana --hosted freezed
```
Points: 40/120.
```sh
git clone https://github.com/rrousselGit/freezed.git
pana freezed/packages/freezed
```
Points: 100/120.

### PR DESCRIPTION
**Pull request by isoos**
Fixes #1089

### CODE REVIEW

File: lib/src/sdk_env.dart [hunk 1]
```diff
@@ -468,18 +468,36 @@ class ToolEnvironment {
...
     FutureOr<R> Function() fn,
   ) async {
-    final backup = await _stripAndAugmentPubspecYaml(packageDir);
+    final now = DateTime.now();
+    final pubspecBackup = await _stripAndAugmentPubspecYaml(packageDir);
+
+    // backup of pubspec_overrides.yaml
+    final poyamlFile = File(p.join(packageDir, 'pubspec_overrides.yaml'));
+    File? poyamlBackup;
+    if (await poyamlFile.exists()) {
```
Comment (sigurdm) [inline lib/src/sdk_env.dart:488]: Usually the "-sync" methods are both faster and more convenient
```suggestion
    if (poyamlFile.existsSync()) {
```

**Review by sigurdm — APPROVED** [review]
Ok for now - but I don't like all this messing around with the files of the package. It seems fragile to me.

Moving forward we should really either:
* Work on a copy of the package
* Make a dummy-package depending on the package.

**Comment by isoos** [comment]
> Work on a copy of the package

We already do the analysis on a copy:
https://github.com/dart-lang/pana/blob/master/lib/src/package_analyzer.dart#L96-L98

> Make a dummy-package depending on the package

Yeah, that would also be nice.

File: lib/src/sdk_env.dart (hunk 1 above)
Comment (isoos) [reply lib/src/sdk_env.dart:488]: Keeping it for consistency, and also we don't want to block the analysis VMs (yet, until sandboxing at least).

**Comment by sigurdm** [comment]
> We already do the analysis on a copy: [...]

Nice - I had forgotten!

**Pull request merged**